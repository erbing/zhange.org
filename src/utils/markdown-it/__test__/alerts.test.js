import MarkdownIt from 'markdown-it'
import alertsPlugin from '../alerts'

const md = new MarkdownIt()
md.use(alertsPlugin())

test('main', () => {
  const html = md.render(`
> __Info__: This is a \`info\`!

> __Warning__: This is a <code>warning</code>!

> __Success__: This is __ok__!

>   __Note__: This is just a _note_!
`)
  expect(html).toMatchSnapshot()
})

test('no matching', () => {
  const html = md.render(`
> __Info__ : hehe

> nah
`)

  expect(html).toMatchSnapshot()
})

test('multiple paragraph', () => {
  const html = md.render(`
> __Info__: hehe
>
> nah
  `)

    expect(html).toMatchSnapshot()
})
