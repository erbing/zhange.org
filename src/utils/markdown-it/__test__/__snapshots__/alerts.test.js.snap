// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main 1`] = `
"<div class=\\"alert info\\">
<p><alert-icon name=\\"alert\\" /></strong>This is a <code>info</code>!</p>
</div>
<div class=\\"alert warning\\">
<p><alert-icon name=\\"alert\\" /></strong>This is a &lt;code&gt;warning&lt;/code&gt;!</p>
</div>
<div class=\\"alert success\\">
<p><alert-icon name=\\"check\\" /></strong>This is <strong>ok</strong>!</p>
</div>
<div class=\\"alert note\\">
<p><alert-icon name=\\"alert\\" /></strong>This is just a <em>note</em>!</p>
</div>
"
`;

exports[`multiple paragraph 1`] = `
"<div class=\\"alert info\\">
<p><alert-icon name=\\"alert\\" /></strong>hehe</p>
<p>nah</p>
</div>
"
`;

exports[`no matching 1`] = `
"<blockquote>
<p><strong>Info</strong> : hehe</p>
</blockquote>
<blockquote>
<p>nah</p>
</blockquote>
"
`;
